src/Crank/CoreWCFPerfClient/Program.cs

```diff
@@ -0,0 +1,155 @@
+    {
+        public const string Binding = "binding";
+        public const string ServiceUrl = "serviceurl";
+        public const string ReportingUrl = "reportingurl";
+        public const string PerfMeasurementDuration = "perfmeasurementduration";
+        public const string Requests = "requests";
+        public const string CloseAndOpen = "closeandopen";
+    }
+
+    public enum TestBinding { BasicHttp, NetTcp }
+
+    class Program
+    {
+        private TestBinding _paramBinding = TestBinding.BasicHttp;
+        private TimeSpan _paramPerfMeasurementDuration = s_defaultPerfMeasurementDuration;
+        private string _paramServiceUrl = "";
+        private int _paramRequests = 0;
+        private bool _paramCloseAndOpen = false;
+
+        private readonly static TimeSpan s_defaultPerfMeasurementDuration = TimeSpan.FromSeconds(10);
+
+        static void Main(string[] args)
+        {
+
+            Program test = new Program();
+
+            if (test.ProcessRunOptions(args))
+            {
+                var startTime = DateTime.Now;
+                int requestTime = 0;
+                int request = 0;
+
+                Binding binding = null;
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        binding = new BasicHttpBinding();
+                        break;
+                    case TestBinding.NetTcp:
+                        binding = new NetTcpBinding(SecurityMode.None);
+                        break;
+                    default:
+                        break;
+                }
+
+                TimeSpan measurementDurationPerTime = test._paramPerfMeasurementDuration ;
+               
+                BenchmarksEventSource.Log.Metadata("firstrequest", "max", "max", "First Request (ms)", "Time to first request in ms", "n0");
+                BenchmarksEventSource.Log.Metadata("corewcfperf/requests", "max", "sum", "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Log.Metadata("corewcfperf/rps/max", "max", "sum", "Requests/sec (max)", "Max requests per second", "n0");
+
+                startTime = DateTime.Now;
+                ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                factory.Open();
+                var client = factory.CreateChannel();
+                var stopwatchFirstReq = new Stopwatch();
+                stopwatchFirstReq.Start();
+                var result = client.Hello("helloworld");
+
+                BenchmarksEventSource.Measure("firstrequest", stopwatchFirstReq.ElapsedMilliseconds);
+
+                while (DateTime.Now <= startTime.Add(measurementDurationPerTime))
+                {
+                    var rtnResult = client.Hello("helloworld");
+                    Console.WriteLine(rtnResult);
```

Comment:
Can this be made to be async? The WCF client on .NET Core performs better when used async as the sync code path uses sync over async which uses more threads.